code-challenges.test.js

```diff
@@ -22,13 +38,63 @@ const fibonacciLength2 = 10
 
 // b) Create the function that makes the test pass.
 
+function generateFibonacci(length) {
+  if (length < 1) return []
+  if (length === 1) return [0]
+  
+  let sequence = [0, 1]
+
+  while (sequence.length < length) {
+    let nextNumber = sequence[sequence.length - 1] + sequence[sequence.length - 2]
+    sequence.push(nextNumber)
+  }
+
+  return sequence
+}
+// What is Fibonacci sequence: The Fibonacci sequence is a series of numbers where each number (after the first two) is the sum of the two preceding ones. It usually starts with 0 and 1, and then the subsequent numbers in the sequence are 1, 2, 3, 5, 8, 13, 21, 34, and so on.
+
 // Pseudo code:
+//            Create a function = generateFibonacci (this takes the length parameter)
+//            Length: can return empty list or equal to 1
+//            It initializes a sequence list with the first two Fibonacci numbers (0 and 1).
+//            Create loop: can calculate the next Fibonacci number by adding the last and second to last of the sequence.
```

Comment:
You need to be explaining what the code is doing, not just what the overall function does / what the fibonacci sequence is. If your test was set up with the correct expected output, you would also get a failing test.